File: src/polar.cpp
#include "polar.h"

// Вывод параметров в форматируемый поток
void In(polar &num, std::ifstream &in) {
    in >> num.a >> num.b;
}

void InRnd(polar &num) {
    num.a = (double) (rand() % 10000) / 2000;
    num.b = (double) rand() / (double) 100 - (double) rand() / (double) 100; 
}

// Вывод параметров в форматируемый поток
void Out(polar &num, std::ofstream &out) {
    out << "(" << num.a << "; " << num.b << ") ";
}


double RationalRepr(polar &num) {
    return num.b;
}

File: hw3/src/polar.py

class polar:
    """
        Polar Coordinates
        a - angle
        b - length
    """
    def __init__(self, a, b):

        self.a = float(a)
        self.b = float(b)

    def rational_repr(self):
        return self.b

    def __str__(self):
        return f"({self.a}; {self.b})"
